# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/WebrtcTCPSocketWrapper.cpp
# Commit: b83124cf3a89
# Full Hash: b83124cf3a891091773bcfe2cd15e36155fc5a2d
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Teach WebrtcTCPSocket to use TLS when configured to. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45103
# ==============================================================================

diff -r b879a0044cf3 -r b83124cf3a89 media/mtransport/WebrtcTCPSocketWrapper.cpp
--- a/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:17:18 2019 +0000
+++ b/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:17:24 2019 +0000
@@ -44,20 +44,22 @@
 
 void WebrtcTCPSocketWrapper::AsyncOpen(
     const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
-    const int& aLocalPort, const shared_ptr<NrSocketProxyConfig>& aConfig) {
+    const int& aLocalPort, bool aUseTls,
+    const shared_ptr<NrSocketProxyConfig>& aConfig) {
   if (!NS_IsMainThread()) {
     MOZ_ALWAYS_SUCCEEDS(mMainThread->Dispatch(
         NewRunnableMethod<const nsCString, const int, const nsCString,
-                          const int, const shared_ptr<NrSocketProxyConfig>>(
+                          const int, bool,
+                          const shared_ptr<NrSocketProxyConfig>>(
             "WebrtcTCPSocketWrapper::AsyncOpen", this,
             &WebrtcTCPSocketWrapper::AsyncOpen, aHost, aPort, aLocalAddress,
-            aLocalPort, aConfig)));
+            aLocalPort, aUseTls, aConfig)));
     return;
   }
 
   MOZ_ASSERT(!mWebrtcTCPSocket, "wrapper already open");
   mWebrtcTCPSocket = new WebrtcTCPSocketChild(this);
-  mWebrtcTCPSocket->AsyncOpen(aHost, aPort, aLocalAddress, aLocalPort,
+  mWebrtcTCPSocket->AsyncOpen(aHost, aPort, aLocalAddress, aLocalPort, aUseTls,
                               aConfig->GetLoadInfoArgs(), aConfig->GetAlpn(),
                               dom::TabId(aConfig->GetTabId()),
                               aConfig->GetProxyPolicy());